File: app/src/main/java/com/v2/desafionubank/di/module/FragmentModule.java
package com.v2.desafionubank.di.module;

import android.content.Context;

import com.v2.desafionubank.di.FragmentContext;
import com.v2.desafionubank.ui.BaseFragment;

import dagger.Module;
import dagger.Provides;

/**
 * Created by CaioSChristino on 20/02/2018.
 */

@Module
public class FragmentModule {
    private BaseFragment mBaseFragment;

    public FragmentModule(BaseFragment baseFragment) {
        mBaseFragment = baseFragment;
    }

    @Provides
    @FragmentContext
    Context provideContext() {
        return mBaseFragment.getContext();
    }

    @Provides
    BaseFragment provideFragment() {
        return mBaseFragment;
    }
}

File: app/src/main/java/com/v2/desafionubank/di/module/FragmentModule.kt
package com.v2.desafionubank.di.module

import android.content.Context
import com.v2.desafionubank.ui.BaseFragment
import dagger.Provides
import com.v2.desafionubank.di.FragmentContext
import dagger.Module


/**
 * Created by csanchez on 19/04/2018.
 */

@Module
class FragmentModule(private val mBaseFragment: BaseFragment) {

    @Provides
    @FragmentContext
    internal fun provideContext(): Context {
        return mBaseFragment.context
    }

    @Provides
    internal fun provideFragment(): BaseFragment {
        return mBaseFragment
    }
}